geolocator_android/android/src/main/java/com/baseflow/geolocator/GeolocatorPlugin.java

```diff
@@ -73,7 +91,8 @@ public static void registerWith(io.flutter.plugin.common.PluginRegistry.Registra
 
     LocationServiceHandlerImpl locationServiceHandler = new LocationServiceHandlerImpl();
     locationServiceHandler.startListening(registrar.context(), registrar.messenger());
-    locationServiceHandler.setActivity(registrar.activity());
+    locationServiceHandler.setContext(registrar.activity());
+    geolocatorPlugin.bindForegroundService(registrar.activity());
```

Comment:
I am aware this is all deprecated but non the less it might be better to use the `registrar.activeContext()` in this case. The `activeContext()` method will return the `activity` if it's available and otherwise will return the Application's context (official documentation can be found [here](https://api.flutter.dev/javadoc/io/flutter/plugin/common/PluginRegistry.Registrar.html#activeContext())).
```suggestion
    locationServiceHandler.setContext(registrar.activeContext());
    geolocatorPlugin.bindForegroundService(registrar.activeContext());
```